packages/blade/src/components/Badges/_decisions/decisions.md

```diff
@@ -0,0 +1,77 @@
+# Badge <!-- omit in toc -->
+Badges are used to show small amount of color coded metadata, which are ideal for getting users attention.
+
+<img src="./badge-thumbnail.png" width="380" />
+
+## Design
+- [Figma - Badge](https://www.figma.com/file/jubmQL9Z8V7881ayUD95ps/Blade---Payment-Light?node-id=8110%3A417)
+
+
+## API
+
+| Prop | Type | Default | Description | Required |
+|---|---|---|---|---|
+| children | `string` | `undefined` | The label to be rendered within the badge. | ✅ |
+| variant | `positive`,`negative`, `notice`, `information`, `neutral` | `neutral` | The variant of the badge to be rendered. |  |
+| contrast | `low`, `high` | `low` | The contrast of the badge to be rendered. |  |
+| size | `medium`, `small` | `medium` | The size of the badge to be rendered. |  |
+| icon | `Icon` | `undefined` | The Blade `Icon` component to be rendered for the badge. |  |
+| fontStyle | `italic`, `normal` | `normal` | The font style of the badge's font. |  |
+| fontWeight | `bold`, `regular` | `regular` | The font weight of the badge's font. |  |
+
+### Sample Usage
+```jsx
+import { Badge } from '@razorpay/components';
+
+<Badge 
+  variant='positive'
+  contrast='high'
+  icon={InfoIcon} 
```

Comment:
<!-- Questions are appropriate if you have a potential concern but are not quite sure if it's relevant or not. Asking the author for clarification or investigation can lead to a quick resolution.  -->
**nit(non-blocking):** How are we importing InfoIcon here? Can we add the import for that here since we only allow Blade icons here?‏

Reply:
<!-- Questions are appropriate if you have a potential concern but are not quite sure if it's relevant or not. Asking the author for clarification or investigation can lead to a quick resolution.  -->
**question:** ‏How and why are we restricting this component to use only Blade icons? Given that the variant is a free-form prop, shall the icon also be a free-form prop here?

Reply:
Added import for icon. We want consumers to use Blade Icons. Generally, we support all feather icons, but if there are any missing and the requirement comes in, we can add the icon directly to blade. 

Icons also have to adhere to a design language - for instance, we only support monochrome icons right now. Making it free-form could lead to unintended usage

Reply:
Understood and agreed. So how are we enforcing the Icons to only be BladeIcons here? Seems to be guidelines right now 🤔

Reply:
Yup. Guidelines and usage examples. We thought about adding it as `iconName` and letting the consumer pass the icon name instead of the component but that would affect tree-shaking negatively. Adding guidelines seem to be the best approach. We still need to figure out a better way to handle icons across components but this seemed the best tradeoff for now